osu.Game.Tests/Gameplay/TestSceneStoryboardSamples.cs

```diff
@@ -70,6 +73,37 @@ public void TestSamplePlaybackAtZero()
             AddUntilStep("sample playback succeeded", () => sample.LifetimeEnd < double.MaxValue);
         }
 
+        [Test]
+        public void TestSamplePlaybackWithRateMods()
+        {
+            GameplayClockContainer gameplayContainer = null;
+            TestDrawableStoryboardSample sample = null;
+
+            OsuModDoubleTime doubleTimeMod = null;
```

Comment:
Why not test more than one mod per-case, along with mods that have custom `SpeedChange` values set. (using `[TestCase(...)]`s)

Reply:
ModRateAdjust and ModTimeRamp both have SpeedChange properties but I would require some type checking in the test to be able to change their values as a test case.
Is there a reason why ModTimeRamp doesn't inherit ModRateAdjust?

Reply:
It would be definitively out of scope, and I would tend to disagree with adding that inheritance.